File: two pointers/Two Sum II.py
def twoSum(numbers: list[int], target: int) -> list[int]:
    l, r = 0, len(numbers) - 1

    while l <= r:
        added = numbers[l] + numbers[r]
        if added > target:
            r -= 1
        elif added < target:
            l += 1
        else:
            return [l, r]

    return []

File: two pointers/Two Sum II.cs
public class SolutionTwoSum
{
    public int[] TwoSum(int[] numbers, int target)
    {
        int l = 0;
        int r = numbers.Length - 1;

        while (l < r)
        {
            int sum = numbers[l] + numbers[r];
            if (sum > target)
                --r;
            else if (sum < target)
                ++l;
            else
                return new int[2] { ++l, ++r };
        }

        return new int[2] { -1, -1 };
    }
}
